integration-tests/specs/payer.test.ts

```diff
@@ -0,0 +1,129 @@
+import {getCurrentSessionId} from "../helpers/session";
+import {getCurrentPlatform} from "../helpers/platform";
+import {getAttributesNameByCurrentPlatform} from "../helpers/attributes";
+
+const COMMON_ATTRIBUTES_NAME = getAttributesNameByCurrentPlatform()
+
+const platform = getCurrentPlatform();
+
+
+const validateSpanPayee = (sessionPayloads, attributesToFind) => {
+  expect(sessionPayloads.Spans.length).toBe(1);
+  const {
+    data: {spans},
+  } = sessionPayloads.Spans[0];
+
+  const itemCountersSpansRequest: SpanEventExpectedRequest = {
+    "emb-session": {
+      expectedInstances: 1,
+      attributes: attributesToFind,
+    },
+  };
+
+  const itemCountersSpansResponse = countSpanEvent(
+    spans,
+    itemCountersSpansRequest,
+  );
+
+  Object.values(itemCountersSpansResponse).forEach(({request, response}) => {
+    expect(response.found).toBe(request.expectedInstances);
+  });
+};
+
+const validateUserPayerForAndroid = (
+  sessionPayloads,
+  currentSessionId,
+  payeeCleaned?,
+) => {
+  expect(sessionPayloads.Events.length).toBe(2);
+  sessionPayloads.Events.forEach(({event, application, user}) => {
+    const {state, session_id, name} = event;
+    expect(state).toBe("foreground");
+    expect(session_id).toBe(currentSessionId);
+    if (!application && name === "_startup") {
+      expect(!!user).toBe(true);
+      const {personas} = user;
+      expect(personas.includes("payer")).toBe(!payeeCleaned);
+    }
+  });
+  const attributesToFind = {
+    "emb.usage.set_user_as_payer": "1",
+  };
+  attributesToFind[COMMON_ATTRIBUTES_NAME.session_id] = currentSessionId
+  if (payeeCleaned) {
+    attributesToFind["emb.usage.clear_user_as_payer"] = "1";
+  }
+  validateSpanPayee(sessionPayloads, attributesToFind);
+};
+
+const validateUserPayerForIOS = (
+  sessionPayloads,
+  currentSessionId,
+  payeeCleaned?,
+) => {
+  expect(sessionPayloads.Events.length).toBe(0);
+  const payeeToValidate = {}
+
+  payeeToValidate[COMMON_ATTRIBUTES_NAME.session_id]=currentSessionId
+
+  validateSpanPayee(sessionPayloads, payeeToValidate);
+
+  expect(sessionPayloads.Spans.length).toBe(1);
+
+  const {
+    metadata: {personas},
+  } = sessionPayloads.Spans[0];
+
+  const [persona] = personas;
+
+  if (payeeCleaned) {
+    expect(persona).toBe(undefined);
+  } else {
+    expect(persona).toBe("payer");
+  }
+};
+const VALIDATE_PAYEE_FUNCTIONS = {
+  android: validateUserPayerForAndroid,
+  iOS: validateUserPayerForIOS,
+};
+describe("Session data - Payer", () => {
+  it("should display the payer data", async () => {
+    const currentSessionId = await getCurrentSessionId(driver);
+
+    const setUserAsPayer = await driver.$("~ADD PAYER");
+    await setUserAsPayer.click();
+
+    await new Promise(r => setTimeout(r, 5000));
+
+    // This is needed because spand are not being sent if I terminate the app
+    await driver.execute("mobile: backgroundApp", {seconds: 1});
```

Comment:
I added a helper around this here https://github.com/embrace-io/embrace-react-native-sdk/pull/53/files#diff-c98bad4c452cdd53e54cd3be82c308920f13d2a02100f9db877fc527c233397c that I'll try to get merged shortly and can be made use of